### PR TITLE
chore(frontend): Fix all typescript errors

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

I want to create a few npm packages to export parts of the posthog app, like `@posthog/apps-common` or `@posthog/lemons` or `@posthog/a-single-insight`. The last one has been asked by customers.

However, code with errors is hard to bundle. 

## Changes

This PR fixes the remaining TypeScript errors in our codebase and removes the "ignore `*Type.ts` files when checking for errors" exception.

After this, `bin/check-typescript-strict` returns 0 errors. If typegen still says something like "Found 2 errors", then you have some old `*Type.ts` files weren't deleted when the `logic.ts` file itself was deleted.

I fixed this by running `yarn typegen:clean`. This however made changes to another 61 files (added `type` to `import type`), which I'm not including in this PR, but in a [followup](https://github.com/PostHog/posthog/pull/10093).

## How did you test this code?

Fixed the errors one by one, saw the nr of errors go down.

### CODE REVIEW

File: bin/check-typescript-strict [hunk 1]
```diff
@@ -3,21 +3,17 @@
 echo "Refreshing Kea logic types and running the TypeScript compiler in dry mode..."
 echo
 
-IGNORED_SPECS=('Type.ts') # *Type.ts* files are ignored, as they are generally created by kea-typegen
```
Comment (Twixes) [inline bin/check-typescript-strict:6]: Considering that really the only purpose of this script was making it possible to ignore certain errors, can we now just remove it and update the `typescript:check` script to be `yarn typegen:write && tsc --strict`? Would make the workflow a bit simpler to grasp

**Comment by Twixes** [comment]
Overall great cleanup!

**Comment by mariusandra** [comment]
CI errors fixed.

image

... and added a small update to `yarn typescript:check`:
- added `--noEmit`, so if someone ever changes things and we do emit in `tsconfig.json`, nothing will be emitted when just checking
- added `"echo 'hey, actual output that means all worked'"` as well

**Comment by mariusandra** [comment]
... except for whatever the `$favourite_damned_place` this is:

image

**Review by Twixes — APPROVED** [review]
🦾

**Pull request merged**